core/src/main/java/io/questdb/cairo/TableReader.java

```diff
@@ -367,10 +374,24 @@ public int getPartitionIndexByTimestamp(long timestamp) {
         return end / PARTITIONS_SLOT_SIZE;
     }
 
+    public long getPartitionMaxTimestampFromMetadata(int partitionIndex) {
+        long minTimestamp = getPartitionMinTimestampFromMetadata(partitionIndex);
+        // todo: the max timestamp must be within the given partition
```

Comment:
Can we do anything now to resolve this todo? If not, would be great to expand the comment with more details and a potential solution. As far as I understand, the problem is to do with split partitions, i.e. the partitions to be squashed later on. If so, we could pick up min timestamp from the next partition (if it's present) and then use `min(txFile.getNextLogicalPartitionTimestamp(minTimestamp) - 1, nextPartitionMinTimestamp - 1)` as a more precise approx max timestamp here.

Reply:
i left this for the merge into nanos branch, we will resolve it there via driver

Reply:
fixed logic here: https://github.com/questdb/questdb/pull/5939/commits/6732acd66746c91856f385e210134bdee08b2093

thanks @ideoma for the explanation